uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrls.spec.js

```diff
@@ -1288,6 +1288,137 @@
         });
       });
 
+      describe('getStatuses', function () {
+        var status = {};
+
+        beforeEach(function () {
+          var collectionName = 'hrleaveandabsences_leave_request_status';
+          var allStatuses = optionGroupMock.getCollection(collectionName);
+
+          status = _.indexBy(allStatuses, 'name');
+
+          initTestController({
+            contactId: CRM.vars.leaveAndAbsences.contactId,
+            userRole: 'admin'
+          });
+
+          $ctrl._init();
+          $ctrl.request = {
+            status_id: null
+          };
+          $ctrl.requestStatuses = status;
+        });
+
+        it('defines a getStatuses method', function () {
+          expect(typeof $ctrl.getStatuses).toBe('function');
+        });
+
+        it('returns an empty array if .request is not defined', function () {
+          delete $ctrl.request;
+
+          var result = $ctrl.getStatuses();
+          expect(result).toEqual([]);
+        });
+
+        it('returns an empty array if .requestStatuses is empty', function () {
+          $ctrl.requestStatuses = {};
+
+          var result = $ctrl.getStatuses();
+          expect(result).toEqual([]);
+        });
+
+        it('returns *More Information Required, Approve* when previous status was not defined', function () {
+          var result = $ctrl.getStatuses();
+
+          expectStatusToBeEqual(result, [
```

Comment:
Why do you need this `expectStatusToBeEqual` function, you can just do
`expect(result).toEqual([the array])`